File: python/q14.py
#!/usr/bin/python3


def collatz(n):
    counter=0
    while n>1:
        if n%2==0:
            n=n/2
        else:
            n=(3*n)+1
        counter+=1
    return counter

runningN=0
runningCollatz=0
n=1
while n<1000000:
    value=collatz(n)
    if value>runningCollatz:
        runningCollatz=value
        runningN=n
    n+=1
print(runningN)

File: go/q14.go
package main
import("fmt")

func collatz(n int) int{
	var counter int = 0
	for n>1 {
		if n%2==0{
			n = n/2
		} else {
			n = (3*n) + 1
		}
		counter += 1
	}
	return counter
}

func main() {
	var runningN int = 0
	var runningCollatz int = 0
	var n int = 1
	for n<1000000 {
		var value int = collatz(n)
		if value>runningCollatz{
			runningCollatz = value
			runningN = n
		}
		n += 1
	}
	fmt.Println(runningN)
}
